lib/dynamoid/criteria/chain.rb

```diff
@@ -176,6 +176,27 @@ def count
         end
       end
 
+      # Returns the first item matching the criteria.
+      #
+      #   Post.where(links_count: 2).first
+      #
+      # Applies `record_limit(1)` to ensure only a single record is fetched
+      # when no non-key conditions are present.
+      #
+      # If used without criteria it just returns the first item of some
+      # arbitrary order.
+      #
+      #   Post.first
+      #
+      # @return [Model|nil]
+      def first
+        if @key_fields_detector.non_key_present?
+          all.to_a.first
```

Comment:
Method `all` returns lazy enumerator to iterate over paginated result. So it makes sense not to materialize the whole collection with calling `to_a` method here.

Reply:
delegating to `super` now